File: TodoSystem/ClientApp/src/app/todo/Models/todoItem.ts
export class TodoItem {
  Id: number | undefined;
  Name: string | undefined;
  Date: Date | undefined;
  Status: StatusEnum | undefined;

  constructor(
    public id: number,
    public name: string,
    public date: Date,
    public status: StatusEnum
  ) { }

}

export enum StatusEnum {
  Done,
  Pending,
  Overdue
}

File: TodoSystem/Models/TodoItem.cs
using System.ComponentModel.DataAnnotations;
using TodoSystem.Enums;

namespace TodoSystem.Models
{
    public class TodoItem
    {
        public int Id { get; set; }
        [Required]
        [MaxLength(30)]
        public string Name { get; set; }
        public DateTime Date { get; set; }
        public StatusEnum Status { get; set; }
    }
}
